src/modules/navigator/mission_feasibility_checker.cpp

```diff
@@ -83,142 +84,33 @@ MissionFeasibilityChecker::checkMissionFeasible(const mission_s &mission,
-			    missionitem.nav_cmd != NAV_CMD_DO_TRIGGER_CONTROL &&
-			    missionitem.nav_cmd != NAV_CMD_DO_DIGICAM_CONTROL &&
-			    missionitem.nav_cmd != NAV_CMD_IMAGE_START_CAPTURE &&
-			    missionitem.nav_cmd != NAV_CMD_IMAGE_STOP_CAPTURE &&
-			    missionitem.nav_cmd != NAV_CMD_VIDEO_START_CAPTURE &&
-			    missionitem.nav_cmd != NAV_CMD_VIDEO_STOP_CAPTURE &&
-			    missionitem.nav_cmd != NAV_CMD_DO_MOUNT_CONFIGURE &&
-			    missionitem.nav_cmd != NAV_CMD_DO_MOUNT_CONTROL &&
-			    missionitem.nav_cmd != NAV_CMD_DO_SET_ROI &&
-			    missionitem.nav_cmd != NAV_CMD_DO_SET_ROI_LOCATION &&
-			    missionitem.nav_cmd != NAV_CMD_DO_SET_ROI_WPNEXT_OFFSET &&
-			    missionitem.nav_cmd != NAV_CMD_DO_SET_ROI_NONE &&
-			    missionitem.nav_cmd != NAV_CMD_DO_SET_CAM_TRIGG_DIST &&
-			    missionitem.nav_cmd != NAV_CMD_DO_SET_CAM_TRIGG_INTERVAL &&
-			    missionitem.nav_cmd != NAV_CMD_SET_CAMERA_MODE &&
-			    missionitem.nav_cmd != NAV_CMD_DO_VTOL_TRANSITION) {
-				takeoff_first = false;
-
-			} else {
-				takeoff_first = true;
-
-			}
-		}
-	}
-
-	if (_navigator->get_takeoff_required() && _navigator->get_land_detected()->landed) {
-		// check for a takeoff waypoint, after the above conditions have been met
-		// MIS_TAKEOFF_REQ param has to be set and the vehicle has to be landed - one can load a mission
-		// while the vehicle is flying and it does not require a takeoff waypoint
-		if (!has_takeoff) {
-			mavlink_log_critical(_navigator->get_mavlink_log_pub(), "Mission rejected: takeoff waypoint required.");
-			return false;
-
-		} else if (!takeoff_first) {
-			// check if the takeoff waypoint is the first waypoint item on the mission
-			// i.e, an item with position/attitude change modification
-			// if it is not, the mission should be rejected
-			mavlink_log_critical(_navigator->get_mavlink_log_pub(), "Mission rejected: takeoff not first waypoint item");
-			return false;
-		}
-	}
-
-	// all checks have passed
-	return true;
+	/*
+	 * Perform check and issue feedback to the user
+	 * Mission is only marked as feasible if takeoff check passes
+	 */
+	return checkTakeoff(mission, home_alt);
 }
 
 bool
-MissionFeasibilityChecker::checkFixedwing(const mission_s &mission, float home_alt, bool home_alt_valid,
-		bool land_start_req)
+MissionFeasibilityChecker::checkFixedwing(const mission_s &mission, float home_alt, bool land_start_req)
 {
-	/* Perform checks and issue feedback to the user for all checks */
-	bool resTakeoff = checkFixedWingTakeoff(mission, home_alt, home_alt_valid);
-	bool resLanding = checkFixedWingLanding(mission, land_start_req);
-
-	/* Mission is only marked as feasible if all checks return true */
-	return (resTakeoff && resLanding);
+	/*
+	 * Perform checks and issue feedback to the user for all checks
+	 * Mission is only marked as feasible if all checks return true
+	 */
+	return (checkTakeoff(mission, home_alt) && checkFixedWingLanding(mission, land_start_req));
```

Comment:
This is lazy evaluation here. We need to do it like it was before so that we generate all mavlink statustext messages.

Reply:
Done. Thanks for pointing this